server/src/business-layer/services/StripeService.ts

```diff
@@ -35,25 +130,39 @@ export default class StripeService {
     })
   }
 
+  /**
+   *
+   * @param client_reference_id generally firebase id
+   * @param email ideally extracted from user JWT
+   * @param return_url
+   * @param line_item format `{price: <price id of item>, quantity: X}`
+   * @param metadata KVP of metadata
+   * @param expires_after_mins Must be over 30 and under 24 hours (1140) we default to 31 minutes
+   * @returns client secret
+   */
   public async createCheckoutSession(
     client_reference_id: string,
     return_url: string,
     line_item: {
       price: string
       quantity: number
     },
-    metadata: Record<string, string>
+    metadata: Record<string, string>,
+    customer_id: string,
+    expires_after_mins: number = 31
   ) {
     const session = await stripe.checkout.sessions.create({
```

Comment:
The docs for [`sessions.create`](https://docs.stripe.com/api/checkout/sessions/object#checkout_session_object-customer_email) mention that for the `customer_email`
> If provided, this value will be used when the Customer object is created. If not provided, customers will be asked to enter their email address. Use this parameter to prefill customer data if you already have an email on file. To access information about the customer once the payment flow is complete, use the customer attribute.

We're currently not providing the `customer_email` here. It's not a problem I think, but might mean that the session email might be different than the firebase email. This probably can end up being the case anyway (e.g., if a user has a different billing email to the UASC email) -- just something to keep in the back of the mind.

Reply:
Actually good point raised here - in the `createNewUser()` for stripe service we force them to use their firebase email but that's probably not the best choice. I'll remove that from the method.